### ISSUE TITLE
Config to handle derail

### ISSUE THREAD

**Issue by quentfr21** [EternalBlueFlame/Traincraft-5#356]
hello

if you can and if it's possible , can you make a config to choose if rollingstock should derail or not

thanks in advance

**Comment by traso56** [comment]
i would add the option to change the derailing speed and maybe a speed escalator since trains are slowish even compared to vanilla horses :(

**Comment by EternalBlueFlame** [comment]
Yeah, what I planned is to have is a set of modes such as "hardcore", "Normal", and "Arcade".
Hardcore is more realistic, you're not gonna make it around vanilla sized turns and you'll be hard pressed to even use vanilla-sized diagonals.
Normal is more like TC now but a bit more realistic, you can take vanilla curves but you'll have to slow down dependant on curve size.
Arcade is almost exactly like TC now, where you'll derail at curves at a certain speed regardless of size.
This system, like everything else, will also be modular so I can extend it as time goes on.

Now regardless of what mode you set it to, trains and rollingstock can also be put into "creative mode" where they don't derail at any speed.

In the current code the creative mode works, but nothing else is implemented yet, there's still a lot of things in progress, but thankfully most of them are simple like this.
